virtual_rainforest/models/animals/functional_group.py

```diff
@@ -0,0 +1,54 @@
+"""The `models.animals.functional_group` module contains a class that organizes
+constants and rate equations used by AnimalCohorts in the
+:mod:`~virtual_rainforest.models.animals` module
+"""  # noqa: D205, D415
+
+from virtual_rainforest.models.animals.constants import (
+    CONVERSION_EFFICIENCY,
+    DAMUTHS_LAW_TERMS,
+    ENDOTHERM_METABOLIC_RATE_TERMS,
+    FAT_MASS_TERMS,
+    INTAKE_RATE_TERMS,
+    MUSCLE_MASS_TERMS,
+)
+
+
+class FunctionalGroup:
+    """This is a class of animal functional groups.
+
+    The goal of this class is to collect the correct constants and scaling relationships
+    needed by an animal cohort such that they are accessed at initialization and stored
+    in the AnimalCohort object as attributes. This should result in a system where an
+    animal cohort can be auto-generated with a few keywords and numbers but that this
+    procedure only need run once, at initialization, and that all further references to
+    constants and scaling relationships are accessed through attributes of the
+    AnimalCohort in question.
+
+
+
+    Needs:
+        - auto-generation system
+        - ability to manually populate specific constants
+
+    """
+
+    def __init__(self, name: str, taxa: str, diet: str) -> None:
+        """The constructor for the FunctionalGroup class."""
+        self.name = name
```

Comment:
Ultimately, I think this would be better having the actual terms as the signature (as in that `dataclass` example I posted).  That way you can create a `FunctionalGroup` instance by specifying all the terms. But... then your current `__init__` method can become a factory method:

```python
class FunctionalGroup:
    ...
    @classmethod
    def from_keywords(cls, name, diet, taxa) -> FunctionalGroup:
          # Contents of current __init__ into a constructor
          return cls(
               name=name, 
               diet=diet, 
               taxa=taxa,
               metabolic_rate_terms = ENDOTHERM_METABOLIC_RATE_TERMS[taxa],
               ...
               )
```

That sets up the two ways of populating them (totally bespoke values versus presets). So I think that is better, but not necessarily for this PR.